app/models/user.rb

```diff
@@ -79,4 +75,8 @@ def username_not_fake_admin
       end
     end
   end
+
+  def ensure_community_user!
+    community_user || create_community_user(reputation: SiteSetting['NewUserInitialRep'])
```

Comment:
Where's `create_community_user` defined for this line?

Reply:
That's a [built in rails helper](https://guides.rubyonrails.org/association_basics.html#methods-added-by-has-one) - for creating a has_one association.

Reply:
Ah, missed that there was a has_one as well as the has_many - I'd been expecting `community_users.create`. :+1: